client/client.go

```diff
@@ -1222,12 +1561,81 @@ func (c *Client) runHandleInvocation(msg *wamp.Invocation) {
 			})
 			return
 		}
-		c.sess.SendCtx(c.ctx, &wamp.Yield{
-			Request:     reqID,
-			Options:     wamp.Dict{},
-			Arguments:   result.Args,
-			ArgumentsKw: result.Kwargs,
-		})
+
+		options := result.Options
+
+		if options == nil {
+			options = wamp.Dict{}
+		}
+
+		message := &wamp.Yield{
+			Request: reqID,
+			Options: options,
+		}
+
+		// Check and handle Payload PassThru Mode
+		// @see https://wamp-proto.org/wamp_latest_ietf.html#name-payload-passthru-mode
+		if pptScheme, _ := options[wamp.OptPPTScheme].(string); pptScheme != "" {
+			// Let's check: was ppt feature announced by dealer?
+			if !c.sess.HasFeature(wamp.RoleDealer, wamp.FeaturePayloadPassthruMode) {
+				// It's protocol violation, so we need to abort connection
+				abortMsg := wamp.Abort{Reason: wamp.ErrProtocolViolation}
+				abortMsg.Details = wamp.Dict{
+					"error": ErrPPTNotSupportedByRouter.Error(),
+				}
+				abortMsg.Details[wamp.OptMessage] = "Peer is trying to use Payload Passthru Mode while it was not announced during HELLO handshake"
```

Comment:
it looks really weird when it's formatted this way, try:
```suggestion
				abortMsg := wamp.Abort{
					Reason: wamp.ErrProtocolViolation}
					Details: wamp.Dict{
						"error": ErrPPTNotSupportedByRouter.Error(),
						wamp.OptMessage: "Peer is trying to use Payload Passthru Mode while it was not announced during HELLO handshake",
					},
				}
```
is there no defined value for `"error"` ? maybe you could add `wamp.OptError` ?

Reply:
Yeah, got it. Agree.